src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1066,7 +1075,7 @@ private void VisitObjectOrDynamicObjectCreation(BoundExpression node, BoundExpre
             if ((object)type != null)
             {
                 bool isTrackableStructType = EmptyStructTypeCache.IsTrackableStructType(type);
-                if (type.IsReferenceType || isTrackableStructType)
+                if (!type.IsValueType || isTrackableStructType)
```

Comment:
>IsValueType [](start = 26, length = 11)

Not related to this PR: I see that we use `.IsReferenceType` in a number of remaining places in `NullableWalker`. It may be good to take a peek to see if they might need to be switched too.